src/sqlfluff/rules/capitalisation/CP05.py

```diff
@@ -50,13 +50,8 @@ class Rule_CP05(Rule_CP01):
             "data_type",
         }
     )
-    _target_elems: List[Tuple[str, str]] = [
-        ("parenttype", "data_type"),
-        ("parenttype", "datetime_type_identifier"),
-        ("parenttype", "primitive_type"),
-        ("type", "data_type_identifier"),
-    ]
-    _exclude_elements: List[Tuple[str, str]] = []
+    _exclude_types = ()
```

Comment:
It looks like the exclusion moved farther down in the rule, around line 89. Any concerns about that? Anything worth adding a comment about?

Reply:
For this rule, I don't think we're changing the logic at all.

- The clause starting `# Skip if not an element of the specified type/name` wasn't being called at all, because it was matching against `_exclude_elements`, which was being set to an empty list. I've just removed it.
- The second check, I've just refactored - no change in logic.
- As for `_exclude_types` and `_exclude_parent_types`, they will never actually be called by `CP05`, because it overrides `_eval` and then only calls `_handle_segment` from the parent which doesn't use those checks. I'm setting them to empty here more as a tidy up. This last point is probably worth stating in a comment - I'll add that.